File: ecbs-os/backend/src/ECBS.Application/ScreenData/AnalysisData.cs
namespace ECBS.Application.ScreenData;

public sealed record CurrentAnalysisData(
    IReadOnlyList<AnalysisKpi> Kpis,
    IReadOnlyList<AnalysisTableRow> AssetRows,
    IReadOnlyList<string> Insights,
    string Message,
    string SiteName,
    string State,
    string UpdatedAt);

public sealed record LiveDataScreenData(
    IReadOnlyList<AnalysisKpi> Kpis,
    IReadOnlyList<AnalysisTableRow> PhaseRows,
    IReadOnlyList<AnalysisTableRow> DeviceRows,
    IReadOnlyList<AnalysisTableRow> AlarmRows,
    IReadOnlyList<AnalysisSummaryRow> SystemRows,
    string ClientName,
    string Message,
    string ProjectName,
    string SiteName,
    string State,
    string UpdatedAt);

public sealed record AnalysisKpi(string Detail, string Icon, string Label, string Tone, string Value);

public sealed record AnalysisTableRow(IReadOnlyList<string> Cells);

public sealed record AnalysisSummaryRow(string Label, string Value);

File: ecbs-os/frontend/src/lib/analysisData.ts
export type AnalysisKpi = {
  detail: string;
  icon: string;
  label: string;
  tone: string;
  value: string;
};

export type AnalysisTableRow = {
  cells: string[];
};

export type AnalysisSummaryRow = {
  label: string;
  value: string;
};

export type CurrentAnalysisData = {
  assetRows: AnalysisTableRow[];
  insights: string[];
  kpis: AnalysisKpi[];
  message: string;
  siteName: string;
  state: string;
  updatedAt: string;
};

export type LiveDataScreenData = {
  alarmRows: AnalysisTableRow[];
  clientName: string;
  deviceRows: AnalysisTableRow[];
  kpis: AnalysisKpi[];
  message: string;
  phaseRows: AnalysisTableRow[];
  projectName: string;
  siteName: string;
  state: string;
  systemRows: AnalysisSummaryRow[];
  updatedAt: string;
};

